### PR TITLE
Docs for using CUDA-enabled docker images

### PR DESCRIPTION
**Pull request by strickvl**
I added a core page which describes how to update your pipeline / step definitions to use a parent base image with CUDA installed. I updated orchestrator and step operator pages to reflect this new addition.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/advanced-guide/pipelines/gpu-hardware.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+```python
+docker_settings = DockerSettings(parent_image="pytorch/pytorch:latest", requirements=["zenml==0.20.5", "torchvision"])
+
+@pipeline(settings={"docker": docker_settings})
+def my_pipeline(...):
+    ...
+```
+
+Adding these two extra settings options and applying them to whatever resource
+is appropriate for your pipelines and use cases will be enough to ensure that
```
Comment (fa9r) [inline docs/book/advanced-guide/pipelines/gpu-hardware.md:53]: somewhat hard to read sentence

File: docs/book/advanced-guide/pipelines/gpu-hardware.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+```python
+docker_settings = DockerSettings(parent_image="pytorch/pytorch:latest")
+
+@pipeline(settings={"docker": docker_settings})
+def my_pipeline(...):
+    ...
+```
+
+2. Add ZenML as an explicit pip requirement
```
Comment (fa9r) [inline docs/book/advanced-guide/pipelines/gpu-hardware.md:37]: why is this necessary, don't we do this under the hood when building images? Or, asked differently, why don't we need to do this when running on CPU?

**Review by fa9r — APPROVED** [review]
LGTM!

File: docs/book/advanced-guide/pipelines/gpu-hardware.md (hunk 2 above)
Comment (strickvl) [reply docs/book/advanced-guide/pipelines/gpu-hardware.md:37]: Not entirely sure, but this is what @schustmi told me this morning. Maybe he has the answer. I think step #2 is just precautionary and perhaps not essential?

File: docs/book/advanced-guide/pipelines/gpu-hardware.md (hunk 1 above)
Comment (strickvl) [reply docs/book/advanced-guide/pipelines/gpu-hardware.md]: Fixed now.

File: docs/book/advanced-guide/pipelines/gpu-hardware.md (hunk 2 above)
Comment (htahir1) [reply docs/book/advanced-guide/pipelines/gpu-hardware.md:37]: You always need ZenML installed in every case, because the code that runs inside the docker image is an entrypoint file which is inside ZenML.That code actually kicks off the pipeline.

In the `CPU` case, you dont specify a parent_image @fa9r, so you the default zenml base image is used which has zenml already installed. With this current approach, a pytorch base image with cuda is being used and of course that means zenml needs to be installed. So its quite mandatory

File: docs/book/advanced-guide/pipelines/gpu-hardware.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+1. Specify a CUDA-enabled parent image in your `DockerSettings`
+
+For full details, see the whole section where we explain how to do this [on the
+containerization page](../../advanced-guide/pipelines/containerization.md). As
+an example, if you wanted to use the latest CUDA-enabled official PyTorch image
+for your entire pipeline run, you could include the following code:
+
+```python
+docker_settings = DockerSettings(parent_image="pytorch/pytorch:latest")
```
Comment (htahir1) [inline docs/book/advanced-guide/pipelines/gpu-hardware.md:30]: Are you sure `pytorch/pytorch:latest` has CUDA installed? Maybe specify a CUDA-specific image to make this very explicit.

**Review by htahir1 — CHANGES_REQUESTED** [review]
Cool - I think Im hoping for a more in-depth analysis of this CUDA business and I left my thoughts in the review. One thing I would like to see on top of my comments is the changing of `examples` of step operators to include the methodology mentioned

File: docs/book/advanced-guide/pipelines/gpu-hardware.md (hunk 3 above)
Comment (strickvl) [reply docs/book/advanced-guide/pipelines/gpu-hardware.md:30]: I am sure, but I can specify something directly so that it's obvious maybe.

File: docs/book/advanced-guide/pipelines/gpu-hardware.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+The core cloud operators all offer prebuilt Docker images that fit with their
+hardware. You can find more information on them here:
+
+- [AWS](https://github.com/aws/deep-learning-containers/blob/master/available_images.md)
+- [GCP](https://cloud.google.com/deep-learning-vm/docs/images)
+- [Azure](https://learn.microsoft.com/en-us/azure/machine-learning/concept-prebuilt-docker-images-inference)
+
+Not all of these images are available on DockerHub, so your orchestrator node
+would have to have permission, for example, to pull images from registries if
+you were to use one of those.
```
Comment (htahir1) [inline docs/book/advanced-guide/pipelines/gpu-hardware.md:78]:
```suggestion
Not all of these images are available on DockerHub, so your please ensure
that the orchestrator environment your pipeline runs in have to have sufficient 
permission to pull images from registries if you were to use one of those.
```

**Review by htahir1 — CHANGES_REQUESTED** [review]
Nice, thanks for the quick turnaround! The examples update is still left and i left one more suggestion
